### PR TITLE
Ruby course: Add example results for two projects

### LINKED ISSUE #28318: Ruby: Provide instructions on testing certain projects
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

I've noticed that a recent change to the Ruby course content includes instructions on testing the HashMap you create in the HashMap project. I thought this was a great addition, but feel as though it'd be a good idea to consistently give info on doing this for other projects where it's missing. Specifically:

- Recursion
- Linked Lists
- Knight Travails

This, of course, becomes irrelevant to add in projects further down when RSpec is introduced officially later in the course. Still I feel like it might be better to offer consistency in testing the projects (Perhaps providing the "Live preview" button on the projects I mentioned as well?) prior to getting into that. 

Would love to discuss thoughts on this!

### Path

Ruby / Rails

### Lesson Url

All the projects I listed.

### (Optional) Discord Name

delldasharky

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by Grayvox**
## Because

Recursion and Linked Lists both lack any examples of inputs and expected outputs to test the student's solutions for the projects, so this PR aims to fix that. Originally, as discussed in the issue linked below, this was going to add it to Knight Travails as well. However, I felt with what method examples are already present in the project that it was probably unnecessary.

Also some minor linting changes to Introduction to Git because they popped up during development and I thought I might as well fix them.

## This PR

- Adds a "Test it out" section to Recursion and Linked Lists
- Adds inputs and expected outputs into the new section

## Issue

Closes #28318

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Many thanks! Some initial comments/changes needed.

Once these are done and a maintainer more familiar with Ruby is happy with this PR's additions and has approved and merged, I'll do a translation for the JS/Node side.

**Comment by Grayvox** [comment]
@MaoShizhong Done! Thanks for that - only linting issues left are what should be resolved in the accessible link text issue

File: ruby/computer_science/project_linked_lists.md [hunk 1]
```diff
@@ -56,11 +56,34 @@ Build the following methods in your linked list class:
...
 1. `#insert_at(value, index)` that inserts a new node with the provided `value` at the given `index`.
 1. `#remove_at(index)` that removes the node at the given `index`.
 
 **Extra Credit Tip:** When you insert or remove a node, consider how it will affect the existing nodes. Some of the nodes will need their `#next_node` link updated.
 
+#### Test it out
+
+Let's test out the Linked List you made!
+
+1. Create a `main.rb` file and make sure it inherits the file containing `LinkedList`. This is where we'll test the list.
```
Comment (JoshDevHub) [inline ruby/computer_science/project_linked_lists.md:70]:
```suggestion
1. Create a `main.rb` file and make sure it requires the file containing `LinkedList`. This is where we'll test the list.
```
Recommend changing this from "inherits" (which generally refers to subclassing) to "requires" (which is the Ruby terminology for loading different files)

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
Just have one small suggestion but otherwise this looks great.

Another small sidebar: I can see that you made this PR from the `main` branch of your fork. This is *very* dangerous. If someone else works on this file and has their PR merged to the curriculum while this is still in review, a "merge conflict" may happen, and you might not have an easy way to resolve it from this PR. So just a reminder that when you're working on something, be sure to create a *new* branch to do the PR with. You'll be guaranteed to avoid a situation where you throw away a bunch of work and have to redo the whole PR (I've seen this happen to people multiple time!). Just a warning for the future.

**Comment by Grayvox** [comment]
Thanks @JoshDevHub ! So sorry about that, and thank you for pointing it out to me - I'll be sure to do that in the future to prevent any issues that might come up. I committed the change you suggested and it should be all ready to go.

**Review by JoshDevHub — APPROVED** [review]
Looks great! Thank you for making this change @Grayvox :rocket:

And thank you for the previous review @MaoShizhong :raised_hands:

**Comment by MaoShizhong** [comment]
I'll get a JS translation up in a bit

**Pull request merged**